### PR TITLE
make safe for govcloud

### PR DESCRIPTION
**Pull request by kilbergr**
This PR makes it safe to use GovCloud with this module by adding partitions. Updates out of date configs as well.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,11 +1,13 @@
+data "aws_partition" "current" {}
```
Comment (mdawn) [inline main.tf:1]: google led me here: https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/Partitions.html

What does this do beyond limit service & region on a role?

Comment (kilbergr) [reply main.tf:1]: It doesn't actually limit service/region on a role--what this is doing is checking for which region you are in (GovCloud is considered a region). Lower in the code, you can see me using the value that was determined here to put the role in the appropriate region. If I try to access a role that is not in the govcloud-accessible region from govcloud, I get errors :(

Comment (mdawn) [reply main.tf:1]: TIL

Comment (cblkwell) [reply main.tf:1]: Actually, it's *not* region. It's which side of the commercial/GovCloud divide you're on. See the first bit about ARNs in the GovCloud guide in the Engineering Playbook. Regions in GovCloud *are* different, but partition is a difference on *top* of that.

Comment (kilbergr) [reply main.tf:1]: @cblkwell I was thinking of it like this: https://aws.amazon.com/about-aws/whats-new/2011/08/16/announcing-aws-govcloud-us/ But it sounds like now that there are multiple US GovCloud regionS, that's not an appropriate definition. I want to clarify so I can send to Ren. 
Does this sound better: GovCloud is a partition with multiple regions. If you are not aligned on using both the GovCloud partition and a GovCloud region, you'll get errors. In this case, I am in the GovCloud region without specifying the GovCloud partition.

Comment (cblkwell) [reply main.tf:1]: Correct. GovCloud has two regions (`us-gov-west-1` and `us-gov-east-1`). So we need to make sure we're using both a GovCloud region and the GovCloud partition type.

**Pull request merged**